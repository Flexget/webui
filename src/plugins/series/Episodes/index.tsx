import React, { FC, useCallback, useMemo } from 'react';
import { useParams, useHistory } from 'react-router';
import { Formik } from 'formik';
import { useInjectPageTitle, useSetAppBarIcon } from 'core/layout/AppBar/hooks';
import { Direction } from 'utils/query';
import { useMergeState } from 'utils/hooks';
import { ArrowBackIos } from '@material-ui/icons';
import { EpisodeContainer } from '../hooks/episodes';
import { GetEpisodeOptions } from '../types';
import Header from './Header';
import EpisodeList from './EpisodeList';
import { useGetShowDetail } from '../hooks/shows';

interface Match {
  showId: string;
}

const EpisodesPage: FC = () => {
  const showId = parseInt(useParams<Match>().showId, 10);

  const { push } = useHistory();

  const { show } = useGetShowDetail(showId);
  const title = show ? `${show.name} - Episodes` : 'Episodes';

  const onClick = useCallback(() => {
    push('/series');
  }, [push]);

  const icon = useMemo(
    () => ({
      Component: ArrowBackIos,
      onClick,
      label: 'go back'
    }),
    [onClick],
  );

  useInjectPageTitle(title);
  useSetAppBarIcon(icon);

  const [options, setOptions] = useMergeState<GetEpisodeOptions>({
    page: 0,
    perPage: 30,
    order: Direction.Desc,
  });

  const handleSubmit = useCallback(values => setOptions(values), [setOptions]);

  return (
    <EpisodeContainer.Provider>
      <Formik initialValues={options} onSubmit={handleSubmit}>
        <Header options={options} />
      </Formik>
      <EpisodeList show={show} options={options} />
    </EpisodeContainer.Provider>
  );
};

export default EpisodesPage;
