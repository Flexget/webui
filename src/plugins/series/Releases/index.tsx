import React, { FC } from 'react';
import { useInjectPageTitle } from 'core/layout/AppBar/hooks';
import { ReleaseContainer } from '../hooks/releases';

const ReleasesPage: FC = () => {
  useInjectPageTitle('Releases');

  return (
    <ReleaseContainer.Provider>
      <div />
    </ReleaseContainer.Provider>
  );
};

export default ReleasesPage;
