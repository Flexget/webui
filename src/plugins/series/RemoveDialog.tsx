import React, { FC, FormEvent, useCallback } from 'react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  DialogContentText,
  Theme,
} from '@material-ui/core';
import { css } from '@emotion/core';
import { RequestState, APIRequest } from 'core/api';

interface Props {
  state: RequestState;
  request: APIRequest;
  open?: boolean;
  onClose: () => void;
  name: string;
}

const errorStyle = (theme: Theme) => css`
  color: ${theme.palette.error.main};
  text-align: center;
`;

const RemoveDialog: FC<Props> = ({
  open = false,
  onClose,
  request,
  state: { error, loading },
  name,
}) => {
  const handleSubmit = useCallback(
    async (e: FormEvent<HTMLFormElement>) => {
      e.preventDefault();
      const resp = await request();
      if (resp.ok) {
        onClose();
      }
    },
    [onClose, request],
  );
  return (
    <Dialog open={open} onClose={onClose} fullWidth maxWidth="xs">
      <form onSubmit={handleSubmit}>
        <DialogTitle>Remove {name}</DialogTitle>
        <DialogContent>
          {error && <DialogContentText css={errorStyle}>{error.message}</DialogContentText>}
          <DialogContentText>Are you sure you would like to remove this {name}?</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onClose} type="button">
            Cancel
          </Button>
          <Button type="submit" color="primary" disabled={loading}>
            Remove
          </Button>
        </DialogActions>
      </form>
    </Dialog>
  );
};

export default RemoveDialog;
