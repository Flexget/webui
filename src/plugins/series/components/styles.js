import styled from '@emotion/styled';
import Grid from '@material-ui/core/Grid';

export const GridCard = styled(Grid)`
  width: 100vw;
`;
