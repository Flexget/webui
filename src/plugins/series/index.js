import { createAsyncComponent } from 'utils/loading';
import registry from 'core/registry';
import saga from './data/saga';
import reducer from './data/reducer';

export default () => registry.registerPlugin('series', {
  component: createAsyncComponent(() => import('plugins/series/components')),
  routeDisplayName: 'Series',
  routeIcon: 'tv',
  reducer,
  saga,
});
