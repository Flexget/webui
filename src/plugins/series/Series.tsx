import React, { FC } from 'react';
import { Route, Switch, useRouteMatch } from 'react-router';
import { useInjectPageTitle } from 'core/layout/AppBar/hooks';
import Shows from './Shows';
import Episodes from './Episodes';
import Releases from './Releases';

const Series: FC = () => {
  useInjectPageTitle('Series');
  const { path } = useRouteMatch();

  return (
    <Switch>
      <Route exact path={path} component={Shows} />
      <Route exact path={`${path}/:showId`} component={Episodes} />
      <Route exact path={`${path}/:showId/epsiodes/:episode`} component={Releases} />
    </Switch>
  );
};

export default Series;
