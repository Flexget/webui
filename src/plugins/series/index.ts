import { createAsyncComponent } from 'utils/loading';
import { registerPlugin } from 'core/plugins/registry';
import LiveTvIcon from '@material-ui/icons/LiveTv';

export default () =>
  registerPlugin('/series', {
    component: createAsyncComponent(() =>
      import(
        /* webpackChunkName: 'SeriesPlugin' */
        'plugins/series/Series'
      ),
    ),
    routeDisplayName: 'Series',
    routeIcon: LiveTvIcon,
  });
