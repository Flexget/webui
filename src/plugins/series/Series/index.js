import { connect } from 'react-redux';
import actions from '../state/shows/actions';
import Series from './Series';

export function mapStateToProps({ series }) {
  return {
    shows: series.shows.items,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getShows: payload => dispatch(actions.getShows.request(payload)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Series);
export { Series };
