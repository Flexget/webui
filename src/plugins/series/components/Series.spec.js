import { mapStateToProps } from 'plugins/series/components';

describe('plugins/series/components/Series', () => {
  it('should return the right stuff', () => {
    expect(mapStateToProps({
      series: {
        shows: {
          items: [],
        },
      },
    })).toMatchSnapshot();
  });
});
