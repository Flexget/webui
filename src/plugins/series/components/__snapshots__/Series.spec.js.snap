// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/series/components/Series should return the right stuff 1`] = `
Object {
  "shows": Array [],
}
`;
